bin/stop_fe.sh

```diff
@@ -66,12 +66,47 @@ pidfile=$PID_DIR/fe.pid
 
 if [ -f $pidfile ]; then
     pid=`cat $pidfile`
-    pidcomm=`ps -p $pid -o comm=`
-    if [ "java"x != "$pidcomm"x ]; then
-        echo "ERROR: pid process may not be FE. "
+
+    # ============================================================
+    # Validation phase: Verify process before any kill operation
+    # All checks must pass before proceeding to kill
+    # ============================================================
+
+    # Step 1: Check if process exists (kill -0 only checks, doesn't kill)
+    if ! kill -0 $pid > /dev/null 2>&1; then
+        echo "Process $pid does not exist, removing stale pid file"
+        rm $pidfile
+        exit 0
+    fi
+
+    # Step 2: Get process information
+    pidcomm=`ps -p $pid -o comm= 2>/dev/null`
+    pidcmd=`ps -p $pid -o command= 2>/dev/null | head -1`
+
+    # Step 3: Verify we can get process information
+    if [ -z "$pidcomm" ] || [ -z "$pidcmd" ]; then
+        echo "ERROR: Cannot get process information for pid $pid"
```

Comment:
can remove the pidfile in error case, since the process info is not as expected, the FE process must be dead and the pid get reused.